internal/repository/sync_test.go

```diff
@@ -202,4 +202,65 @@ func TestClone(t *testing.T) {
 		_, err = Clone(context.Background(), remoteTmpDir, dirName, "")
 		assert.ErrorIs(t, err, ErrDirExists)
 	})
+
+	t.Run("successful clone without specifying dir, with trailing slashes in repository path.", func(t *testing.T) {
+		localTmpDir := t.TempDir()
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
+
+		repo, err := Clone(context.Background(), remoteTmpDir+"//", "", "")
+		assert.Nil(t, err)
+		head, err := repo.r.Head()
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, commitID, head.Hash())
+
+		localRSLRef, err := repo.r.Reference(plumbing.ReferenceName(rsl.Ref), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remoteRSLRef.Hash(), localRSLRef.Hash())
+		localPolicyRef, err := repo.r.Reference(plumbing.ReferenceName(policy.PolicyRef), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remotePolicyRef.Hash(), localPolicyRef.Hash())
+	})
+
+	t.Run("successful clone with dir. With trailing slashes in repository path.", func(t *testing.T) {
```

Comment:
This will always succeed, because here the remoteURL is a (temp.) directory path. Go-git handles trailing slashes in directory path, but does not handle two trailing slashes in http url as git does. Should I clone [ci-demo](https://github.com/gittuf/ci-demo) in a temp dir to test that?

Reply:
Interesting. I want to minimize the number of "live" tests that make a network call, so I'm leaning no. I suggest we just submit a patch to go-git.

Reply:
This applies to both of the new tests right? If we aren't really testing any behavior here at the moment, let's take it out.

Reply:
Done.

Reply:
> Interesting. I want to minimize the number of "live" tests that make a network call, so I'm leaning no. I suggest we just submit a patch to go-git.

I think that handling more than single slash is a trivial matter. Even git does not handle more than two slashes, while other commands such as curl handles any number of slashes. In any case, our tests do not fail, so I think we can submit a patch later on, if we need to.